### ISSUE TITLE
Demo page not working during local development

### ISSUE THREAD

**Issue by NickyMeuleman** [BenDMyers/showmy.chat#61]
Pages like http://localhost:8080/c/showmychat?theme=default&DEMO=true remain unpopulated.
The CSS gets applied, as you can see the background change when you select a different theme, but messages never show up.

The errors in the browser console:
![image](https://user-images.githubusercontent.com/30179461/149637400-9dd9b8ef-7f36-4c03-9850-01ee1ce4afe1.png)

I think what's happening is the errors in the console stop JS executions, and the script that should post faked messages never gets a chance to run.

<details>
<summary>
the console errors is textform:
</summary>
Access to fetch at 'https://api.betterttv.net/3/cached/users/twitch/762083631' from origin 'http://localhost:8080' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'https://localhost:8080' that is not equal to the supplied origin. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
bttvIntegration.js:80 GET https://api.betterttv.net/3/cached/users/twitch/762083631 net::ERR_FAILED 404
defaultFetch	@	bttvIntegration.js:80
getBttvChannelEmoteDict	@	bttvIntegration.js:67
init	@	handleChat.js:228
await in init (async)		
(anonymous)	@	handleChat.js:233
bttvIntegration.js:80 
 
 Uncaught (in promise) TypeError: Failed to fetch
    at defaultFetch (bttvIntegration.js:80:25)
    at getBttvChannelEmoteDict (bttvIntegration.js:67:32)
    at init (handleChat.js:228:24)
defaultFetch	@	bttvIntegration.js:80
getBttvChannelEmoteDict	@	bttvIntegration.js:67
init	@	handleChat.js:228
await in init (async)		
(anonymous)	@	handleChat.js:233
</details>

**Comment by BenDMyers** [comment]
I'm wondering if this is failing for accounts that don't have BTTV support set up. Does it work if you use `FrostyyPaws` as the channel name?

**Comment by NickyMeuleman** [comment]
for http://localhost:8080/c/FrostyyPaws&DEMO=true
![image](https://user-images.githubusercontent.com/30179461/149637966-43b3d62f-3ee8-47eb-8b36-cff238ca1cfd.png)

**Comment by KenAKAFrosty** [comment]
It's odd that the Access-Control-Allow-Origin is http**s** for local host in your request there, meanwhile the actual origin is just http because it's localhost. I don't have the same error and can run demo fine locally, so it seems related to your browser sending that header that way?  I'm a little murky on the details on how or why it would do that

**Comment by NickyMeuleman** [comment]
To make previews work again during development, I added this at the top of the `defaultFetch` function in `scripts/bttvIntegration.js`

```
if (window.location.hostname == "localhost") {
	return new Promise((resolve) => {
		resolve({});
	});
}
```

**Comment by BenDMyers** [comment]
The above snippet was merged in with #62. It'll get rid of the errors and prevent crashes, but since it also kind of nullifies any efforts to test/validate BTTV locally, I'm going to leave this issue open while we investigate any other approaches that might work here.

**Comment by jbolda** [comment]
Just gave this a shot, and I can't seem to create a CORS error. Possible that it was dealt with upstream?
